ddl/column.go

```diff
@@ -125,7 +125,10 @@ func (d *ddl) onAddColumn(t *meta.Meta, job *model.Job) (ver int64, _ error) {
 	// if errorBeforeDecodeArgs {
 	// 	return ver, errors.New("occur an error before decode args")
 	// }
-
+	if err = checkAddColumnTooManyColumns(tblInfo.Columns); err != nil {
```

Comment:
I don't understand. `onAddColumn` will be called 4 times.
My opinion is that `tblInfo.Columns` contains `tblInfo.Columns` and the new column when the new column's state isn't StateNone.
e.g.
tblInfo.Columns is c1, the new column is c2.
At first, c2 is StateNone, so tblInfo.Columns is only c1.
Next, the c2 is StateDeleteOnly, so tblInfo.Columns are c1 and c2.

Reply:
I understand what you mean.